### PR TITLE
Google drive enhancement permissions

### PR DESCRIPTION
**Pull request by tkatzir**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/37655

## Description
Enhance the existing Google Drive integration so we are able to build a playbook that tells Google Drive to revoke a user's permission (permission ID) on a specific file/folder or list of files/folders in Google Drive based on their email address. This helps with DLP use-cases.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py [hunk 1]
```diff
@@ -1,15 +1,23 @@
 from CommonServerPython import *
+import demistomock as demisto
+from GSuiteApiModule import *  # noqa: E402
```
Comment (yaakovi) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py:3]: restore the location of the import

**Review by yaakovi — CHANGES_REQUESTED** [review]
* you are missing release notes (please bump the pack version by a minor)
* one of the unit tests is failing
* some of the linter's fail - to fix the pylint errors regarding the `Resource` class, see other google integration (more specifically, the `.pylintrc` file under the integration's folder).
* only commented on the first appearance but some comments relevant for several places.

**Comment by yaakovi** [comment]
@tkatzir 
regarding the page token output (after a short discussion with @Arsenikr), the context should be as the following:
```
GoogleDrive.<object type> : {
	PageToken: <token>,
	<object type>: [
		{
			<object 1 data>,
		},
		{
			<object 2 data>,
		},
	]
}
```
where `object type` is one of `Drive`, `File`, `Permission`

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py [hunk 2]
```diff
@@ -695,8 +798,843 @@ def drive_changes_list_command(client, args: Dict[str, Any]) -> CommandResults:
...
+    """
+    Download a file from Google Drive
+
+    :param client: Client object.
+    :param args: Command arguments.
+
+    :return: Command Result.
+    """
+
+    demisto.info('google-drive-file-download')
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py]: https://github.com/demisto/content/commit/c5a7ea313ed098d580fb38f4d21aecc06b256b13

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py [hunk 3]
```diff
@@ -695,8 +798,843 @@ def drive_changes_list_command(client, args: Dict[str, Any]) -> CommandResults:
...
+
+    demisto.info('google-drive-file-delete')
+    prepare_file_command_res = prepare_file_command_request(client, args, scopes=COMMAND_SCOPES['FILE_DELETE'])
+    http_request_params = prepare_file_command_res['http_request_params']
+
+    # client.set_authorized_http(scopes=COMMAND_SCOPES['FILE_DELETE'], subject=prepare_file_command_res['user_id'])
+
+    url_suffix = URL_SUFFIX['DRIVE_FILES_ID'].format(args.get('file_id'))
+    client.http_request(url_suffix=url_suffix, method='DELETE', params=http_request_params)
+    # TODO Check the response?
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py]: Obsolete. The response is empty.

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py [hunk 4]
```diff
@@ -695,8 +798,843 @@ def drive_changes_list_command(client, args: Dict[str, Any]) -> CommandResults:
...
+    outputs: Dict = {
+        OUTPUT_PREFIX['FILE']: outputs_context
+    }
+    outputs = GSuiteClient.remove_empty_entities(outputs)
+    readable_hr += files_hr
+
+    return CommandResults(
+        outputs=outputs,
+        readable_output=readable_hr,
+        raw_response=response
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py]: https://github.com/demisto/content/commit/7c12fedecfb08a70413dc61eb6fca1abdf4a1143

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive_test.py [hunk 5]
```diff
@@ -491,3 +491,465 @@ def test_flatten_comment_mentioned_user_keys_for_fetch_incident():
...
+
+    dict_1 = {
+        'key1': 'val1',
+        'key2': 'val2',
+    }
+    dict_2 = {
+        'key3': 'val3',
+    }
+
+    # copy_dict_value(source_dict: dict_1, dest_dict: dict_2, source_dict_key: str, dest_dict_key: str = None):
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive_test.py]: https://github.com/demisto/content/commit/513c7c4dd47dd0f6f2151564c981c49590f93613

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 6]
```diff
@@ -1,26 +1,31 @@
+name: GoogleDrive
 category: IT Services
-commonfields:
-  id: GoogleDrive
-  version: -1
+display: Google Drive
 configuration:
-- display: Fetch incidents
-  name: isFetch
+- name: isFetch
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:5]: https://github.com/demisto/content/commit/d70916f941201ac238b47fe99ae21fc9eaccee63

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 7]
```diff
@@ -1713,17 +1803,1623 @@ script:
...
+    - name: permission_id
+      description: The ID of the permission.
+      required: false
+      isArray: false
+      defaultValue: ""
+      predefined:
+      - ""
+    outputs:
+    - contextPath: kind
+      description: description
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml]: https://github.com/demisto/content/commit/859b8f4f21db42d95742626ab147877e834a9a58

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 8]
```diff
@@ -1713,17 +1803,1623 @@ script:
...
+      required: false
+      isArray: false
+      defaultValue: ""
+      predefined:
+      - ""
+    outputs:
+    - contextPath: GoogleDrive.Drive.kind
+      description: Identifies what kind of resource this is. Value is the fixed string
+        "drive#drive".
+      type: String
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml]: That's the standard in that integration.
I made the description more general:
https://github.com/demisto/content/commit/d876b905cf8712705bbdcb2799fcaf0ccb80545c

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 9]
```diff
@@ -1713,17 +1803,1623 @@ script:
...
+    - contextPath: GoogleDrive.Drive.name
+      description: The name of this shared drive.
+      type: String
+    - contextPath: GoogleDrive.Drive.colorRgb
+      description: The color of this shared drive as an RGB hex string.
+      type: String
+    - contextPath: GoogleDrive.Drive.backgroundImageLink
+      description: A short-lived link to this shared drive's background image.
+      type: String
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml]: https://github.com/demisto/content/commit/9da2a0847361751ff4ea52f3ca380ec7186576c3

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 10]
```diff
@@ -1713,17 +1803,1623 @@ script:
...
+      - ""
+    - name: user_id
+      description: The user's primary email address.
+      required: false
+      isArray: false
+      defaultValue: ""
+      predefined:
+      - ""
+    - name: drive_id
+      description: id of the shared drive.
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml]: https://github.com/demisto/content/commit/9da2a0847361751ff4ea52f3ca380ec7186576c3

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py [hunk 11]
```diff
@@ -249,9 +330,31 @@ def set_true_for_empty_dict(d):
...
 
 
+# def remove_false_values(dict: Dict[str, Any]) -> List[str]:
+#     ret_value = []
+#
+#     for current_item in dict.keys():
+#         if dict[current_item]:
+#             ret_value.append(current_item)
+#
+#     return ret_value
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py]: https://github.com/demisto/content/commit/b0e075ae41746ba688fd31a467de7d9787a11086

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 12]
```diff
@@ -1713,17 +1803,1623 @@ script:
...
+      required: false
+      isArray: false
+      defaultValue: ""
+      predefined:
+      - ""
+    - name: fields
+      description: The paths of the fields you want included in the response.
+      required: false
+      isArray: false
+      defaultValue: kind, id, name, themeId, capabilities, createdTime, hidden, restrictions
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:1961]: I've kept the kind value

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 13]
```diff
@@ -1713,17 +1803,1623 @@ script:
...
+      name: supports_all_drives
+      predefined:
+      - "true"
+      - "false"
+      required: false
+      secret: false
+    outputs:
+    - contextPath: GoogleDrive.File.id
+      description: id of the deleted file.
+      type: String
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:3079]: Removed

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py [hunk 14]
```diff
@@ -695,8 +798,843 @@ def drive_changes_list_command(client, args: Dict[str, Any]) -> CommandResults:
...
+    :param args: Command arguments.
+
+    :return: Command Result.
+    """
+
+    # Specific file
+    prepare_file_read_request_res = prepare_file_read_request(client, args)
+    http_request_params = prepare_file_read_request_res['http_request_params']
+    # Make sure we have the fields "kind" and "id".
+    http_request_params['fields'] = 'kind, id, ' + args.get('fields', 'kind, id')
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py:1010]: Yes, and it's ok. I tested it.

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 15]
```diff
@@ -1713,17 +1803,1623 @@ script:
...
+    - contextPath: GoogleDrive.Drive.name
+      description: The name of this shared drive.
+      type: String
+    - contextPath: GoogleDrive.Drive.colorRgb
+      description: The color of this shared drive as an RGB hex string.
+      type: String
+    - contextPath: GoogleDrive.Drive.backgroundImageLink
+      description: A short-lived link to this shared drive's background image.
+      type: String
+    - contextPath: GoogleDrive.Drive.capabilities.canAddChildren
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:1978]: It's for the user_id from the input.

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py [hunk 16]
```diff
@@ -695,8 +798,843 @@ def drive_changes_list_command(client, args: Dict[str, Any]) -> CommandResults:
...
+    http_request_params['fields'] = '*'
+    url_suffix = URL_SUFFIX['DRIVE_DRIVES_ID'].format(args.get('drive_id'))
+    response = client.http_request(url_suffix=url_suffix, method='GET', params=http_request_params)
+    return handle_response_single_drive(response, args)
+
+
+def handle_response_drive_list(response: Dict[str, Any]) -> CommandResults:
+
+    outputs_context = []
+    readable_hr = ''
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py]: https://github.com/demisto/content/commit/8a7ecef9dd18186dfab8b7f1837e8991a6aaad8c

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py [hunk 17]
```diff
@@ -695,8 +798,843 @@ def drive_changes_list_command(client, args: Dict[str, Any]) -> CommandResults:
...
+
+    :return: Command Result.
+    """
+
+    # Specific file
+    prepare_file_read_request_res = prepare_file_read_request(client, args)
+    http_request_params = prepare_file_read_request_res['http_request_params']
+    # Make sure we have the fields "kind" and "id".
+    http_request_params['fields'] = 'kind, id, ' + args.get('fields', 'kind, id')
+    # TODO Remove empty?
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py]: I need the kind and id fields, and it's ok to have duplicates. I checked it.

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py [hunk 18]
```diff
@@ -695,8 +798,843 @@ def drive_changes_list_command(client, args: Dict[str, Any]) -> CommandResults:
...
+    :param client: Client object.
+    :param args: Command arguments.
+
+    :return: Command Result.
+    """
+
+    demisto.info('google-drive-file-upload')
+    prepare_file_command_res = prepare_file_command_request(client, args, scopes=COMMAND_SCOPES['FILES'])
+    http_request_params = prepare_file_command_res['http_request_params']
+    http_request_params['uploadType'] = 'multipart'
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py]: https://github.com/demisto/content/commit/4d1c997c27427f707ac1c5a9934386f94c746c7b

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py [hunk 19]
```diff
@@ -695,8 +798,843 @@ def drive_changes_list_command(client, args: Dict[str, Any]) -> CommandResults:
...
+    """
+    Upload a file to Google Drive
+
+    :param client: Client object.
+    :param args: Command arguments.
+
+    :return: Command Result.
+    """
+
+    demisto.info('google-drive-file-upload')
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py]: Removed

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 20]
```diff
@@ -1713,17 +1803,1623 @@ script:
...
+      - ""
+    - name: drive_id
+      description: id of the shared drive.
+      required: false
+      isArray: false
+      defaultValue: ""
+      predefined:
+      - ""
+    - name: fields
+      description: The paths of the fields you want included in the response.
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml]: https://github.com/demisto/content/commit/0387843fe1589969bed23e332fcf46b6b77210fe

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py (hunk 1 above)
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py]: https://github.com/demisto/content/commit/77e493dff369e8aa5d500f61d36769941541f40e

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py [hunk 21]
```diff
@@ -695,8 +798,843 @@ def drive_changes_list_command(client, args: Dict[str, Any]) -> CommandResults:
...
+                                  headerTransform=pascalToSpace,
+                                  removeNull=False)
+
+    ret_value = CommandResults(
+        readable_output=table_hr_md,
+    )
+    return ret_value
+
+
+def prepare_body_for_file_upload(args: Dict[str, Any]) -> Dict[str, Any]:
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py]: https://github.com/demisto/content/commit/507f15ffee3a54aaa5503da7dec70d9932d81ae4

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py [hunk 22]
```diff
@@ -695,8 +798,843 @@ def drive_changes_list_command(client, args: Dict[str, Any]) -> CommandResults:
...
+
+def prepare_drives_human_readable(outputs_context: List[Dict[str, Any]]) -> str:
+    """
+    Prepares human readable for google-drives-list command.
+
+    :param outputs_context: Context data.
+
+    :return: Human readable.
+    """
+    drive_hr: List[Dict[str, Any]] = [{}]
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py]: https://github.com/demisto/content/commit/0d5c91b2b26fb812d03010fdab62cf578a4a4081
https://github.com/demisto/content/commit/194d820307dce3bcfffccb8bc95e653f998c4f56

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 23]
```diff
@@ -1713,17 +1803,1623 @@ script:
...
+      predefined:
+      - ""
+    - name: page_token
+      description: Page token for shared drives.
+      required: false
+      isArray: false
+      defaultValue: ""
+      predefined:
+      - ""
+    - name: q
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml]: https://github.com/demisto/content/commit/839f064b6b2083dee1d9e2c51772f2492ff5d533

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py [hunk 24]
```diff
@@ -40,28 +48,101 @@
...
     'DRIVE_CHANGES_LIST': 'GoogleDrive.DriveChange(val.time == obj.time && val.fileId == obj.fileId &&'
                           ' val.driveId == obj.driveId && val.userId == obj.userId)',
     'DRIVE_CHANGES_LIST_PAGE_TOKEN': 'GoogleDrive.PageToken.DriveChange(val.driveId == obj.driveId'
                                      ' && val.userId == obj.userId)',
+
     'DRIVE_ACTIVITY_LIST': 'GoogleDrive.DriveActivity',
-    'DRIVE_ACTIVITY_LIST_PAGE_TOKEN': 'GoogleDrive.PageToken.DriveActivity'
+    'DRIVE_ACTIVITY_LIST_PAGE_TOKEN': 'GoogleDrive.PageToken.DriveActivity',
+
+    'DRIVES_LIST_PAGE_TOKEN': 'GoogleDrive.PageToken',
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py]: https://github.com/demisto/content/commit/55968da82d34e6759c4e24b2fe5a1d496e0ca7f0

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py [hunk 25]
```diff
@@ -695,8 +798,843 @@ def drive_changes_list_command(client, args: Dict[str, Any]) -> CommandResults:
...
+        outputs_context.append(current_drive)
+
+    drives_hr = prepare_drives_human_readable(outputs_context)
+
+    outputs: Dict = {
+        OUTPUT_PREFIX['DRIVE']: outputs_context
+    }
+    if response.get('nextPageToken', ''):
+        outputs[OUTPUT_PREFIX['DRIVES_LIST_PAGE_TOKEN']] = response['nextPageToken']
+        readable_hr += NEXT_PAGE_TOKEN.format(response.get('nextPageToken'))
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py]: https://github.com/demisto/content/commit/55968da82d34e6759c4e24b2fe5a1d496e0ca7f0

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py [hunk 26]
```diff
@@ -695,8 +798,843 @@ def drive_changes_list_command(client, args: Dict[str, Any]) -> CommandResults:
...
+    outputs = GSuiteClient.remove_empty_entities(outputs)
+    readable_hr += drives_hr if response.get('drives', '') \
+        else HR_MESSAGES['NOT_FOUND'].format('Drives')
+
+    return CommandResults(
+        outputs=outputs,
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py:877]: https://github.com/demisto/content/commit/55968da82d34e6759c4e24b2fe5a1d496e0ca7f0

File: Packs/GoogleDrive/TestPlaybooks/playbook-GoogleDrive-Permissions-Test.yml [hunk 27]
```diff
@@ -0,0 +1,625 @@
...
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "15"
+    scriptarguments:
+      data:
+        simple: Hello World
+      filename:
+        simple: HelloWorld
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/TestPlaybooks/playbook-GoogleDrive-Permissions-Test.yml]: https://github.com/demisto/content/commit/740b84aaaee9b38f145c73288b788b5f1dadcc25

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive_test.py [hunk 28]
```diff
@@ -491,3 +491,465 @@ def test_flatten_comment_mentioned_user_keys_for_fetch_incident():
...
+    """
+    mocker_http_request.side_effect = ValueError("SOME_ERROR")
+
+    from GoogleDrive import drives_list_command
+
+    args = {
+        'use_domain_admin_access': True
+    }
+
+    with pytest.raises(Exception, match="SOME_ERROR"):
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive_test.py]: https://github.com/demisto/content/commit/04d0030d30216570aad4ff4bac3bfca172ed1146

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py [hunk 29]
```diff
@@ -695,8 +798,843 @@ def drive_changes_list_command(client, args: Dict[str, Any]) -> CommandResults:
...
+    copy_dict_value(source_dict=args, dest_dict=http_request_params, source_dict_key='q', dest_dict_key='q')
+    copy_dict_value(source_dict=args, dest_dict=http_request_params, source_dict_key='page_size', dest_dict_key='pageSize')
+    copy_dict_value(source_dict=args, dest_dict=http_request_params, source_dict_key='page_token', dest_dict_key='pageToken')
+
+    # user_id can be overridden in the args
+    user_id = args.get('user_id') or client.user_id
+    client.set_authorized_http(scopes=COMMAND_SCOPES['DRIVES'], subject=user_id)
+
+    return {
+        'client': client,
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py]: https://github.com/demisto/content/commit/6ec36cd8616be14f3f79f8b0c5e3e9d01b88a3e5

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive_test.py [hunk 30]
```diff
@@ -491,3 +491,465 @@ def test_flatten_comment_mentioned_user_keys_for_fetch_incident():
...
+    copy_dict_value(source_dict=dict_1, dest_dict=dict_2, source_dict_key='key1')
+    assert dict_2['key1'] == 'val1'
+
+    assert 'key21' not in dict_2
+    copy_dict_value(source_dict=dict_1, dest_dict=dict_2, source_dict_key='key2', dest_dict_key='key_21')
+    assert dict_2['key_21'] == 'val2'
+
+
+@patch(MOCKER_HTTP_METHOD)
+def test_drives_list_command_success(mocker_http_request, gsuite_client):
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive_test.py]: https://github.com/demisto/content/commit/f8db2a96fbb523d3e1d9d7e421d0d937626a7e4c

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 31]
```diff
@@ -1713,17 +1779,1590 @@ script:
...
+    description: Delete a permission.
+    deprecated: false
+    arguments:
+    - name: file_id
+      description: id of the requested file.
+      required: false
+      isArray: false
+      defaultValue: ""
+      predefined:
+      - ""
```
Comment (yaakovi) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:3339]: remove empty predefined and default values
```suggestion
```

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 32]
```diff
@@ -1713,17 +1779,1590 @@ script:
...
+      type: String
+    - contextPath: GoogleDrive.FilePermission.FilePermission.photoLink
+      description: A link to the user's profile photo, if available.
+      type: String
+  - name: google-drive-file-permission-delete
+    description: Delete a permission.
+    deprecated: false
+    arguments:
+    - name: file_id
+      description: id of the requested file.
```
Comment (yaakovi) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:3334]: for any argument of this kind, specify how to find it (which command returns it).

**Review by yaakovi — COMMENTED** [review]
revisit all the unresolved comments and the new one as well

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml (hunk 12 above)
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:1929]: Removed all references to the 'kind' field:
https://github.com/demisto/content/commit/cb2cd4776639379130ac890ad0332e44a96d8321

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py (hunk 1 above)
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.py]: https://github.com/demisto/content/commit/e73f0309c72de8052a37c1437d76b19697b9ea4e

File: Packs/GoogleDrive/pack_metadata.json [hunk 33]
```diff
@@ -1,8 +1,8 @@
 {
     "name": "Google Drive",
-    "description": "Google Drive allows users to store files on their servers, synchronize files across devices, and share files. This integration helps you to create a new drive, query past activity and view change logs performed by the users.",
+    "description": "Google Drive allows users to store files on their servers, synchronize files across devices, and share files. This integration helps you to create a new drive, query past activity and view change logs performed by the users, as well as list drives and files, and manage their permissions.",
     "support": "xsoar",
-    "currentVersion": "1.0.1",
+    "currentVersion": "1.0.2",
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/pack_metadata.json]: https://github.com/demisto/content/commit/7ccdf8b7f3b9eb912ef03b4ed6dd02a99fd156e2

File: Packs/GoogleDrive/ReleaseNotes/1_0_2.md [hunk 34]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+##### Google Drive
+- Updated the Docker image to: *demisto/googleapi-python3:1.0.0.13812*.
+- List drives and files
+- Manage drive and files permissions.
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/ReleaseNotes/1_0_2.md]: https://github.com/demisto/content/commit/7ccdf8b7f3b9eb912ef03b4ed6dd02a99fd156e2

File: Packs/GoogleDrive/Integrations/.pylintrc [hunk 35]
```diff
@@ -0,0 +1,6 @@
+[TYPECHECK]
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/.pylintrc]: https://github.com/demisto/content/commit/7f7597702aa30c068c3233e87c8f7e6a33111927

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 36]
```diff
@@ -1713,17 +1779,1590 @@ script:
...
+  runonce: false
+  script: "-"
+  subtype: python3
+  type: python
+commonfields:
+  id: GoogleDrive
+  version: -1
+fromversion: 5.0.0
+tests:
+- GoogleDrive-Test
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:3284]: https://github.com/demisto/content/commit/c4791fb49479c194dac14d8f52f8176043776f33

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml (hunk 31 above)
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml]: https://github.com/demisto/content/commit/76e0fe16bc8328ca0fd0b119739dbfad05c5af5f

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml (hunk 32 above)
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml]: https://github.com/demisto/content/commit/b6ab42c24c2aa6bf80b6e626853f01917eb0aead

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml (hunk 10 above)
[the remark this answers is not in the document]
Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml]: https://github.com/demisto/content/commit/b6ab42c24c2aa6bf80b6e626853f01917eb0aead

File: Packs/GoogleDrive/ReleaseNotes/1_1_0.md [hunk 37]
```diff
@@ -0,0 +1,23 @@
...
+  - ***google-drive-drive-get***
+  - ***google-drive-files-list***
+  - ***google-drive-file-get***
+  - ***google-drive-file-upload***
+  - ***google***
+  - ***google***
+  - ***google***
+  - ***google***
+  - ***google***
+  - ***google***
```
Comment (yaakovi) [inline Packs/GoogleDrive/ReleaseNotes/1_1_0.md:16]: ?

**Review by yaakovi — APPROVED** [review]
please fix the release notes

**Comment by ShirleyDenkberg** [comment]
@yuvalbenshalom @yaakovi yml file and RN review is completed.  README needs to be regenerated.

File: Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml [hunk 38]
```diff
@@ -1713,17 +1779,1430 @@ script:
...
+- GoogleDrive-Permissions-Test
```
Comment (yaakovi) [inline Packs/GoogleDrive/Integrations/GoogleDrive/GoogleDrive.yml:3208]:
```suggestion
- Google Drive Permissions Test
```

File: Packs/GoogleDrive/Integrations/GoogleDrive/README.md [hunk 39]
```diff
@@ -1,88 +1,30 @@
-Google Drive allows users to store files on their servers, synchronize files across devices, and share files. This integration helps you to create a new drive, query past activity and view change logs performed by the users. 
-This integration was integrated and tested majorly with G Suite Basic and Enterprise edition.
+Google Drive allows users to store files on their servers, synchronize files across devices, and share files. This integration helps you to create a new drive, query past activity, and view change logs performed by the users.
+This integration was integrated and tested with version xx of GoogleDrive
```
Comment (yaakovi) [inline Packs/GoogleDrive/Integrations/GoogleDrive/README.md:2]: remove this line

Comment (tkatzir) [reply Packs/GoogleDrive/Integrations/GoogleDrive/README.md]: https://github.com/demisto/content/pull/13643/commits/a0e95a499d78e6acb52adc2d7faf7af13834a0fb

**Pull request merged**